packages/server/src/AccountsServer.js

```diff
@@ -50,41 +50,71 @@ export class AccountsServer {
       : toUsernameAndEmail({ ...user });
 
     let foundUser;
+
+    if (this.options.passwordAuthenticator) {
+      try {
+        foundUser = await this._externalPasswordAuthenticator(this.options.passwordAuthenticator, id, username, email, password);
+      }
+      catch (e) {
+        throw new AccountsError({ message: e });
+      }
+    }
+    else {
+      foundUser = await this._defaultPasswordAuthenticator(id, username, email, password);
+    }
+
+    if (!foundUser) {
+      throw new AccountsError({ message: 'User not found [403]' });
+    }
+
+    // $FlowFixMe
+    const sessionId = await this.db.createSession(foundUser.id, ip, userAgent);
+    const { accessToken, refreshToken } = this.createTokens(sessionId);
+
+    return {
+      sessionId,
+      user: foundUser,
+      tokens: {
+        refreshToken,
+        accessToken,
+      },
+    };
+  }
+
+  async _externalPasswordAuthenticator(authFn, id, username, email, password) {
+    return await authFn(id, username, email, password, password);
```

Comment:
`password` is being twice in pass.

Consider changing the function arguments to `user: PasswordLoginUserType, password: string` similar to `AccountsServer.loginWithPassword`